client.go

```diff
@@ -224,30 +226,54 @@ func (c *Client) do(method string, path string, body io.Reader, query url.Values
 	}
 	parsed.RawQuery = query.Encode()
 
-	req, err := http.NewRequest(method, parsed.String(), body)
-	if err != nil {
-		return err
-	}
+	return withRetries(
+		func() error {
+			req, err := http.NewRequest(method, parsed.String(), body)
+			if err != nil {
+				return err
+			}
 
-	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.AtlasToken))
-	req.Header.Add("Content-Type", "application/vnd.api+json")
+			req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.AtlasToken))
+			req.Header.Add("Content-Type", "application/vnd.api+json")
 
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
+			resp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				return err
+			}
+			defer resp.Body.Close()
+
+			switch {
+			case resp.StatusCode == 401:
+				return ErrUnauthorized
+			case resp.StatusCode == 404:
+				return ErrNotFound
+			case resp.StatusCode != 200:
+				return ErrBadStatus
+			}
 
-	switch {
-	case resp.StatusCode == 401:
-		return ErrUnauthorized
-	case resp.StatusCode == 404:
-		return ErrNotFound
-	case resp.StatusCode != 200:
-		return ErrBadStatus
-	}
+			decoder := json.NewDecoder(resp.Body)
+			err = decoder.Decode(&recv)
+			return err
+		},
+		func(e error) bool {
+			if err == ErrBadStatus {
+				return true
+			}
+			return false
+		},
+		10,
+	)
+}
 
-	decoder := json.NewDecoder(resp.Body)
-	err = decoder.Decode(&recv)
+func withRetries(f func() error, shouldRetry func(e error) bool, attempts int) error {
+	interval := 10 * time.Millisecond
```

Comment:
maybe increase interval to 1 second? not sure what the rate limits of tfe are

Reply:
I'll meet you in the middle and bump the initial interval to 500ms.  The failures we are receiving right now are not rate limit failures (they are 504's, probably some issue on hashicorp's side).